### PR TITLE
[Flow test] Support flow test with input file

### PR DESCRIPTION
**Pull request by lalala123123**
# Description

```
pf flow test --flow flow/path --inputs input.json
```

```
pf.test(flow="path/to/flow", inputs="path/to/inputs.json")
```

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_pf_client.py [hunk 1]
```diff
@@ -336,6 +337,12 @@ def test(
         :return: The result of flow or node
         :rtype: dict
         """
+        # Load the inputs for the flow test from sample file.
+        if isinstance(inputs, (str, Path)):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_pf_client.py:341]: maybe check the file extension

File: src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py [hunk 2]
```diff
@@ -299,6 +299,30 @@ def test_pf_flow_test(self):
...
+            "--flow",
+            f"{FLOWS_DIR}/web_classification",
+            "--inputs",
+            f"{DATAS_DIR}/webClassification1.jsonl",
+        )
+        output_path = Path(FLOWS_DIR) / "web_classification" / ".promptflow" / "flow.output.json"
+        assert output_path.exists()
+
+        # Test flow test with invalid simple input file
+        with pytest.raises(ValueError) as ex:
```
Comment (wangchao1230) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py:315]: add a test on invalid file extension passed in

File: src/promptflow-devkit/promptflow/_sdk/_pf_client.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_sdk/_pf_client.py:341]: add logic to check file extension

File: src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py:315]: add test case

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**